server/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java

```diff
@@ -884,9 +899,9 @@ protected boolean doEquals(QueryStringQueryBuilder other) {
                 Objects.equals(tieBreaker, other.tieBreaker) &&
                 Objects.equals(rewrite, other.rewrite) &&
                 Objects.equals(minimumShouldMatch, other.minimumShouldMatch) &&
-                Objects.equals(lenient, other.lenient) &&
-                timeZone == null ? other.timeZone == null : other.timeZone != null &&
-                Objects.equals(timeZone.getID(), other.timeZone.getID()) &&
+                Objects.equals(lenient, other.lenient) && 
+                (timeZone == null ? other.timeZone == null
+                        : (other.timeZone != null && Objects.equals(timeZone.getID(), other.timeZone.getID()))) &&
```

Comment:
nit: I find equals impls easier to read when they are symetric, eg. something like this should be correct?
```
Objects.equals(
    timeZone == null ? null : timeZone.getID(),
    other.timeZone == null ? null : otherTimeZone.getID())`